carbon.conf

```diff
@@ -215,14 +215,54 @@ ENABLE_TAGS = False
 # and any other settings you want to customize, defaults are inherited
 # from [carbon] section.
 # You can then specify the --instance=b option to manage this instance
+[cache:1]
+CACHE_QUERY_PORT = 10012
+LINE_RECEIVER_PORT = 10013
+UDP_RECEIVER_PORT = 10013
+PICKLE_RECEIVER_PORT = 10014
+[cache:2]
+CACHE_QUERY_PORT = 10022
+LINE_RECEIVER_PORT = 10023
+UDP_RECEIVER_PORT = 10023
+PICKLE_RECEIVER_PORT = 10024
+[cache:3]
+CACHE_QUERY_PORT = 10032
+LINE_RECEIVER_PORT = 10033
+UDP_RECEIVER_PORT = 10033
+PICKLE_RECEIVER_PORT = 10034
+[cache:4]
+CACHE_QUERY_PORT = 10042
+LINE_RECEIVER_PORT = 10043
+UDP_RECEIVER_PORT = 10043
+PICKLE_RECEIVER_PORT = 10044
+[cache:5]
+CACHE_QUERY_PORT = 10052
+LINE_RECEIVER_PORT = 10053
+UDP_RECEIVER_PORT = 10053
+PICKLE_RECEIVER_PORT = 10054
+[cache:6]
+CACHE_QUERY_PORT = 10062
+LINE_RECEIVER_PORT = 10063
+UDP_RECEIVER_PORT = 10063
+PICKLE_RECEIVER_PORT = 10064
+[cache:7]
+CACHE_QUERY_PORT = 10072
+LINE_RECEIVER_PORT = 10073
+UDP_RECEIVER_PORT = 10073
+PICKLE_RECEIVER_PORT = 10074
+[cache:8]
+CACHE_QUERY_PORT = 10082
+LINE_RECEIVER_PORT = 10083
+UDP_RECEIVER_PORT = 10083
+PICKLE_RECEIVER_PORT = 10084
 
 
 
 [relay]
 LINE_RECEIVER_INTERFACE = 0.0.0.0
-LINE_RECEIVER_PORT = 2013
+LINE_RECEIVER_PORT = 2003
 PICKLE_RECEIVER_INTERFACE = 0.0.0.0
-PICKLE_RECEIVER_PORT = 2014
+PICKLE_RECEIVER_PORT = 2004
```

Comment:
just wondering: are any of these changed + newly added ports needed to be accessible from the outside?
or are they just used for a dispatch mechanism inside the container only?

i.e. does the command to start this image remain the same as before? or will it require addititional port mapping + potentially firewall rules?

Reply:
This pickle port is not used anymore actually.  The whole idea of this PR is that the two ports that can be used to send data to Graphite (2003 and 2004) are now directed to carbon-c-relay, which handles the dispatching work from inside the container.  From the outside perspective, nothing changes at all.